test/jdk/java/net/httpclient/UserAuthWithAuthenticator.java

```diff
@@ -0,0 +1,375 @@
+            var encoded = java.util.Base64.getEncoder().encodeToString(plainCreds.getBytes(US_ASCII));
+            var request = HttpRequest.newBuilder().uri(URI.create(serverMock.baseURL() + "/some_serv_url"))
+                .setHeader("User-Agent", "myUserAgent")
+                .setHeader("Authorization", "Basic " + encoded)
+                .build();
+
+            var response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            assertEquals(200, response.statusCode());
+            assertEquals(data1, response.body());
+
+            var serverStr = serverMock.getRequest(0);
+            assertContains(serverStr, "/some_serv_url");
+            assertPattern(".*^User-Agent:.*myUserAgent.*", serverStr);
+            assertPattern(".*^Authorization:.*Basic " + encoded + ".*", serverStr);
+            System.out.println("testServerOnly: OK");
+        } finally {
+            serverMock.stopMocker();
+        }
+    }
+
+    // This is effectively a regression test for existing behavior
+    static void testServerOnlyAuthenticator() throws IOException, InterruptedException {
+        Mocker serverMock = new Mocker(authenticatorResponses);
+        serverMock.start();
+        try (var client = HttpClient.newBuilder()
+                .version(java.net.http.HttpClient.Version.HTTP_1_1)
+                .authenticator(new ServerAuth())
+                .build()) {
+
+            // credentials set in the server authenticator
+            var plainCreds = "serverUser:serverPwd";
+            var encoded = java.util.Base64.getEncoder().encodeToString(plainCreds.getBytes(US_ASCII));
+            var request = HttpRequest.newBuilder().uri(URI.create(serverMock.baseURL() + "/some_serv_url"))
+                .setHeader("User-Agent", "myUserAgent")
+                .build();
+
+            var response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            assertEquals(200, response.statusCode());
+            assertEquals(data1, response.body());
+
+            var serverStr = serverMock.getRequest(1);
+            assertContains(serverStr, "/some_serv_url");
+            assertPattern(".*^User-Agent:.*myUserAgent.*", serverStr);
+            assertPattern(".*^Authorization:.*Basic " + encoded + ".*", serverStr);
+            System.out.println("testServerOnlyAuthenticator: OK");
+        } finally {
+            serverMock.stopMocker();
+        }
+    }
+
+    static void close(Closeable... clarray) {
+        for (Closeable c : clarray) {
+            try {
+                c.close();
+            } catch (Exception e) {}
+        }
+    }
+
+    static class Mocker extends Thread {
+        final ServerSocket ss;
+        final String[] responses;
+        volatile List<String> requests;
+        volatile InputStream in;
+        volatile OutputStream out;
+        volatile Socket s = null;
+
+        public Mocker(String[] responses) throws IOException {
+            this.ss = new ServerSocket(0);
```

Comment:
Given some of the issues we see in our CIs, I think it would be better to use loopback address for this ServerSocket instead of the default wildcard address.